src/experimental/riff-raff-yaml-file/deployments/lambda.ts

```diff
@@ -2,6 +2,32 @@ import type { GuStack } from "../../../constructs/core";
 import type { GuLambdaFunction } from "../../../constructs/lambda";
 import type { RiffRaffDeployment } from "../types";
 
+interface S3LocationProps {
+  bucketSsmLookup?: boolean;
+  bucketSsmKey?: string;
+  prefixStackToKey?: boolean;
+  prefixAppToKey?: boolean;
+  prefixStageToKey?: boolean;
+}
+
+const locationProps = (lambda: GuLambdaFunction): S3LocationProps => {
+  const bucketProp =
+    lambda.bucketNamePath === undefined
+      ? {
+          bucketSsmLookup: true,
+        }
+      : {
+          bucketSsmKey: lambda.bucketNamePath,
+        };
```

Comment:
I wasn't sure if `bucketSsmLookup: true` is required in both cases. But looking at RR it looks not the case - though this serves as a reminder that we should remove the `bucketSsmLookup` parameter fullstop and just use whichever of bucketName or bucketSsmKey is set (or the default value for the latter if neither).

Reply:
I think this change should be made in a separate PR. Updating in this branch modifies existing behaviour and causes the `riff-raff.yaml` generation tests outside of the scope of this change to fail, (although I do not expect the change to be impactful i'd rather be cautious).